lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -383,50 +700,123 @@ void IMConstPropPass::markBlockExecutable(Block *block) {
       markRegResetOp(regReset);
     else if (auto mem = dyn_cast<MemOp>(op))
       markMemOp(mem);
+    else if (auto subindex = dyn_cast<SubindexOp>(op))
+      markSubindexOp(subindex);
+    else if (auto subfield = dyn_cast<SubfieldOp>(op))
+      markSubfieldOp(subfield);
   }
 }
 
 void IMConstPropPass::markWireOrUnresetableRegOp(Operation *wireOrReg) {
-  // If the wire/reg has a non-ground type, then it is too complex for us to
-  // handle, mark it as overdefined.
-  // TODO: Eventually add a field-sensitive model.
+  // If the wire/reg has a non-ground type, then it is too complex for us
+  // to handle, mark it as overdefined.
   auto resultValue = wireOrReg->getResult(0);
-  if (!resultValue.getType().cast<FIRRTLType>().getPassiveType().isGround())
+  if (!isTrackableType(resultValue.getType()))
     return markOverdefined(resultValue);
 
-  // Otherwise, this starts out as InvalidValue and is upgraded by connects.
-  mergeLatticeValue(resultValue, InvalidValueAttr::get(resultValue.getType()));
+  // Otherwise, this starts out as InvalidValue and is upgraded by
+  // connects.
+  auto &destTypes = getLeafGroundTypes(resultValue.getType());
+  for (unsigned i = 0, e = destTypes.size(); i < e; i++) {
+    auto destType = destTypes[i].cast<FIRRTLType>();
+    mergeLatticeValue({resultValue, i, /*isKnownRoot=*/true},
+                      InvalidValueAttr::get(destType));
+  }
 }
 
 void IMConstPropPass::markRegResetOp(RegResetOp regReset) {
-  // If the reg has a non-ground type, then it is too complex for us to handle,
-  // mark it as overdefined.
-  // TODO: Eventually add a field-sensitive model.
-  if (!regReset.getType().getPassiveType().isGround())
+  // If the reg has a passive type, then it is too complex for us to
```

Comment:
`If the reg has a NON passive type,`. ==> is this actually allowed in the firrtl dialect?  What is an example?

Reply:
Registers have to be passive. This check is probably unnecessary and you can just remove it in a push to `main`.

Reply:
Thank you, it was my misunderstanding.

Reply:
Plz make it an assert though!